src/App.tsx

```diff
@@ -1,24 +1,76 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+interface State {
+  isClockVisible: boolean,
+  clockName: number,
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 0,
+  };
+
+  showClock = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  hideClock = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  setRandomName = () => {
+    this.setState({
+      clockName: Math.ceil(Math.random() * 10),
+    });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1>{`React clock ${clockName}`}</h1>
+
+        <button
+          type="button"
+          onClick={this.setRandomName}
+        >
+          set random name
+        </button>
+
+        <button
+          type="button"
+          onClick={this.showClock}
+        >
+          show clock
+        </button>
+
+        <button
+          type="button"
+          onClick={this.hideClock}
+        >
+          hide clock
+        </button>
+
+        <p>
+          Current time:
+          {' '}
+          {isClockVisible && (
+            <div>
+              <Clock name={clockName} />
+            </div>
```

Comment:
```suggestion
            <Clock name={clockName} />
```
do you need that div?